akka-stream-tests/src/test/java/akka/stream/javadsl/SinkTest.java

```diff
@@ -153,4 +152,17 @@ public void mustBeAbleToConvertToJavaInJava() {
         akka.stream.scaladsl.Sink.cancelled();
     Sink<Integer, NotUsed> javaSink = scalaSink.asJava();
   }
+
+  public void sinkForeachMustBeDocumented()
+      throws InterruptedException, ExecutionException, TimeoutException {
+    // #foreach
+    Sink<Integer, CompletionStage<Done>> printlnSink = Sink.foreach(System.out::println);
+    CompletionStage<Done> cs = Source.from(Arrays.asList(1, 2, 3, 4)).runWith(printlnSink, system);
+    // 1
+    // 2
+    // 3
+    // 4
+    // #foreach
+    assertEquals(Done.done(), cs.toCompletableFuture().get(1, TimeUnit.SECONDS));
```

Comment:
Could perhaps be worth including a callback printing from the completion or failure as well since that is something people often miss out on?